tutorials/applications/QAOA and QAA to solve a QUBO problem.ipynb

```diff
@@ -0,0 +1,622 @@
+    "    \"t\": np.random.uniform(8, 10, LAYERS),\n",
+    "    \"s\": np.random.uniform(1, 3, LAYERS),\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "example_dict = quantum_loop(np.r_[guess[\"t\"], guess[\"s\"]])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can then plot the distribution of the samples, to see the most frequent bitstrings sampled."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def plot_distribution(C):\n",
+    "    C = dict(sorted(C.items(), key=lambda item: item[1], reverse=True))\n",
+    "    indexes = [\"01011\", \"00111\"]  # MIS indexes\n",
+    "    color_dict = {key: \"r\" if key in indexes else \"g\" for key in C}\n",
+    "    plt.figure(figsize=(12, 6))\n",
+    "    plt.xlabel(\"bitstrings\")\n",
+    "    plt.ylabel(\"counts\")\n",
+    "    plt.bar(C.keys(), C.values(), width=0.5, color=color_dict.values())\n",
+    "    plt.xticks(rotation=\"vertical\")\n",
+    "    plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_distribution(example_dict)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The bitstrings `01011` and `00111` (in red) correspond to the two optimal solutions (calculated at the beginning of the notebook). The goal of QAOA is to choregraph interferences between the basis states, in order to maximize the frequency of the MIS states. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Optimization "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We estimate the cost of a sampled state vector by making an average over the samples. This is done by taking the corresponding bitstring ${\\bf z}=(z_1, \\ldots, z_N)$ and calculating\n",
+    "\n",
+    "$$\n",
+    "C({\\bf z}) = {\\bf z}^\\top \\cdot Q \\cdot {\\bf z}\n",
+    "$$\n",
+    "\n",
+    "Determining the cost of a given bitstring takes polynomial time. The average estimate is then used in the classical loop to optimize the variational parameters $\\tau$ and $t$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_cost_colouring(bitstring, Q):\n",
+    "    z = np.array(list(bitstring), dtype=int)\n",
+    "    cost = z.T @ Q @ z\n",
+    "    return cost\n",
+    "\n",
+    "\n",
+    "def get_cost(counter, Q):\n",
+    "    cost = sum(counter[key] * get_cost_colouring(key, Q) for key in counter)\n",
+    "    return cost / sum(counter.values())  # Divide by total samples"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We define `func` in order to do the minimization."
```

Comment:
Something a bit more comprehensive would most likely help the reader to understand the function and its arguments. For example:
‶To perform a minimization loop, we define the following function that will be called at each step by SciPy. `*args` enables to pass the QUBO value, and `params` contains the trial value to score, which changes at each step.″

Reply:
Added the sentence